src/content/developers/docs/smart-contracts/composability/index.md

```diff
@@ -8,7 +8,61 @@ incomplete: true
+
+Smart contract composability generally works off three principles: modularity, autonomy, and discoverability: 
+
+**1. Modularity**: This is the ability of individual components to perform a specific task. In Ethereum, every smart contract has specific use-case (as shown in the Uniswap example).
+
+**2. Autonomy**: Composable components must be able to operate independently. Each smart contract in Ethereum is self-executing and can function without relying on other parts of the system. 
+
+**3. Discoverability**: Developers cannot call external contracts or integrate software libraries into applications if the former are not publicly available. By design, smart contracts are open-source; anyone can call a smart contract or fork a codebase. 
+
+## Benefits of composability {#benefits-of-composability}
+
+### Shorter development cycle
+
+Composability reduces the work that developers have to do when creating [decentralized applications](https://ethereum.org/en/dapps/#what-are-dapps) (dapps). As Naval Ravikant [puts it](https://twitter.com/naval/status/1444366754650656770): "Open source means every problem has to be solved once."
+
+Once a particular smart contract solves one problem, other developers can reuse it, so they don’t have to solve the same problem. In this way, developers can take existing software libraries and add extra functionality to create new dapps. 
+
+### Greater innovation
+
+Composability encourages innovation and experimentation. That’s because developers are free to reuse, modify, duplicate, or integrate open-source code to create desired results. 
+
+Also, development teams spend less time on basic functionality and can allocate more time to experimenting with new features. This explains why there are so many great projects in the Ethereum ecosystem today—developers can focus on improve on existing dapps and build better products. 
+
+### Better user experience
+
+Interoperability between components of the Ethereum ecosystem improves the user experience. Users can access greater functionality when dapps integrate external smart contracts than in a fragmented ecosystem where applications cannot communicate.
+
+## Examples of composability in Ethereum {#composability-in-ethereum}
+
+### Token swaps
+
+Say you have a dapp that requires to pay for transactions in ETH. You can add more payment options by integrating token swap logic from Uniswap contracts. This code will automatically convert the user’s token to ETH before the contract executes the called function.
+
+### Governance
+
+Building bespoke governance systems for a [decentralized autonomous organization](https://ethereum.org/en/dao/) (DAO) can be expensive and time-consuming. However, you can use [Aragon Client](https://client.aragon.org/)—an open-source governance toolkit from Aragon—to bootstrap your DAO and quickly create a usable governance framework. 
```

Comment:
- We should explain utility before names (i.e. it should be written that we should be able to drop the name of any project offering this service in place of Aragon, if we wanted)
- Let's assume they know the DAO acronym at this point

```suggestion
Building bespoke governance systems for a [DAO](/dao/) can be expensive and time-consuming. Instead, you could use an open-source governance toolkit, such as [Aragon Client](https://client.aragon.org/), to bootstrap your DAO to quickly create a governance framework. 
```